lib/Parse/ParseExpr.cpp

```diff
@@ -1927,6 +1927,37 @@ parseStringSegments(SmallVectorImpl<Lexer::StringSegment> &Segments,
       Status |= E;
       if (E.isNonNull()) {
         Exprs.push_back(E.get());
+        
+        if (auto tuple = dyn_cast<TupleExpr>(Exprs.back())) {
```

Comment:
Is the expression ever not a TupleExpr? Maybe use cast<> here instead?

Reply:
The expression actually *should* be a ParenExpr. I don’t know of a reason it should be a TupleExpr without one of the nested conditions being true, but if it ever was, as far as I know it would be valid.